plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/explorerActions/SignInToQAction.kt

```diff
@@ -39,7 +39,7 @@ abstract class SignInToQActionBase(actionName: String) : DumbAwareAction(actionN
         UiTelemetry.click(project, "auth_start_Q")
         val connectionManager = ToolkitConnectionManager.getInstance(project)
         connectionManager.activeConnectionForFeature(QConnection.getInstance())?.let {
-            reauthConnectionIfNeeded(project, it)
+            reauthConnectionIfNeeded(project, it, isReAuth = true)
```

Comment:
is this not an initial login?

Reply:
This checks for an existing Q connection: https://github.com/aws/aws-toolkit-jetbrains/pull/4819/files#diff-899f1e03be46308c47e08c0407c1a1711e9e94c5b9afd38b649231041079bd44R41 and if not, it starts with a new login, hence marked this as reauth

Reply:
no we would not consider exiting creds option as reauth. only when creds expire or user is clicking "reconnect". for existing we can emit a ui_click that would be enough to filter. because if user had sso on AWS toolkit, ideally this is their first login on Amazon Q.

Reply:
Think of `reauth` as flag that will be used to check if users are logging in again on same compute/same product